src/AppBundle/Service/CsvGeneratorService.php

```diff
@@ -0,0 +1,162 @@
+     */
+    private function generateTransactionsCsvLines(ReportInterface $report)
+    {
+        //foreach($report->getBankAccounts() as $bankAccount) {
+        //$this->generateBankAccountSummary($bankAccount);
+        $this->generateCsvHeaders();
+        $this->generateTransactionRows($report->getGifts(), 'gift');
+        $this->generateTransactionRows($report->getExpenses(), 'expenses');
+        $this->generateTransactionRows($report->getMoneyTransactionsOut(), 'money out');
+        $this->generateTransactionRows($report->getMoneyTransactionsIn(), 'money in');
+
+        //}
+    }
+
+    /**
+     * Generates a Bank Account Summary to the CSV file pointer
+     *
+     * @param BankAccount $bankAccount
+     */
+    private function generateBankAccountSummary(BankAccount $bankAccount)
+    {
+        // @todo remove following feedback on whether an account summary is required
+        $summaryFields = [
+            [" "],
+            [" "],
+            ["ACCOUNT SUMMARY"],
+            [$bankAccount->getBank() . "   -   " . ucfirst($bankAccount->getAccountType()) ." account" ],
+            ["**** " . $bankAccount->getAccountNumber() . "      (" . $bankAccount->getSortCode(). ")"],
+            [$bankAccount->getIsJointAccount() ? "JOINT ACCOUNT" : ""],
+            [" "],
+            [" "],
+
+
+        ];
+
+        foreach($summaryFields as $line)
+        {
+            fputcsv($this->fd, $line);
+        }
+    }
+
+    /**
+     * Generate CSV Headers
+     */
+    private function generateCsvHeaders()
+    {
+        $headers = ['Type', 'Category' ,'Amount', 'Bank name', 'Account', 'Description'];
+        fputcsv($this->fd, $headers);
+    }
+
+    /**
+     * Generates Transaction row
+     *
+     * @param $transactions
+     * @param $type
+     */
+    private function generateTransactionRows($transactions, $type)
+    {
+        foreach($transactions as $t) {
+            /** @var $t \AppBundle\Entity\Report\MoneyTransaction */
+            fputcsv(
+                $this->fd, [
+                    ucFirst($type),
+                    (property_exists($t, 'category') ?
+                        $this->translator->trans(
+                            'form.category.entries.' . $t->getCategory().'.label', [], 'report-money-transaction') : ''),
+                    $t->getAmount(),
+                    (!empty($t->getBankAccount()) ? $t->getBankAccount()->getDisplayName() : ""),
+                    (!empty($t->getBankAccount()) ? $t->getBankAccount()->getBank() : ""),
+                    $this->generateDescription($t)
+                ]
+            );
+        }
+    }
+
+    /**
+     * Generates a description. Expenses and Gifts have an 'explanation' property,
+     * Money transactions have a description property.
+     *
+     * @param $transaction
+     * @return string
+     */
+    private function generateDescription($transaction)
+    {
+        if (method_exists($transaction, 'getDescription')) {
```

Comment:
alternative could be using interfaces hasDescription, hasExplnation, add use "instance of" on those. Less risk of typos and string mismatch